features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/utils/DrawableUtils.kt

```diff
@@ -37,27 +41,30 @@ internal class DrawableUtils(
         drawableHeight: Int,
         displayMetrics: DisplayMetrics,
         requestedSizeInBytes: Int = MAX_BITMAP_SIZE_IN_BYTES,
-        config: Config = Config.ARGB_8888
-    ): Bitmap? {
-        val (width, height) = getScaledWidthAndHeight(
-            drawableWidth,
-            drawableHeight,
-            requestedSizeInBytes
-        )
+        config: Config = Config.ARGB_8888,
+        base64SerializerCallback: Base64SerializerCallback,
+        bitmapCreationCallback: Base64Serializer.BitmapCreationCallback
+    ) {
+        val backgroundWork =
+            createBackgroundLambda(drawableWidth, drawableHeight, requestedSizeInBytes, displayMetrics, config)
 
-        val bitmap = getBitmapBySize(displayMetrics, width, height, config) ?: return null
-        val canvas = canvasWrapper.createCanvas(bitmap) ?: return null
+        val foregroundWork = createForegroundLambda(drawable, bitmapCreationCallback)
```

Comment:
why do you need this ? Can't you just use an executor that executes the background work and in the `onSuccess` of that you can go on with your `foregroundWork` ? Also I would make sure I am using the same executor in the `DrawableUtils` and `Base64Serializer` class otherwise you will create too many threads and also we are going to have problems when debugging this code.